### PR TITLE
use stored OD values for factor groups

### PR DESCRIPTION
**Pull request by ThomasBreuer**
... and set obvious `orthogonal_discriminants` values automatically for perfect central extensions

This way, the character table display with ODs works automatically for "2.A8", "2.HS", etc.

### CODE REVIEW

File: experimental/OrthogonalDiscriminants/src/data.jl [hunk 1]
```diff
@@ -112,13 +113,64 @@ function orthogonal_discriminants(tbl::Oscar.GAPGroupCharacterTable)
           res[l[2]] = l[4]
         end
         return res
+      else
+#TODO: compute the reductions mod p (not dividing the group order)
```
Comment (fingolfin) [inline experimental/OrthogonalDiscriminants/src/data.jl:117]: should there be an `error(...)` here, or is it OK if the code "falls through" to whatever comes next?

File: experimental/OrthogonalDiscriminants/src/data.jl [hunk 2]
```diff
@@ -112,13 +113,64 @@ function orthogonal_discriminants(tbl::Oscar.GAPGroupCharacterTable)
...
+    end
+    if facttbl != nothing
+      # Set the known values for the factor.
+      mp = [findfirst(is_equal(restrict(x, tbl)), tbl) for x in facttbl]
+      data = data[string(p)]
+      for l in data
+        res[mp[l[2]]] = l[4]
+      end
+      # Set values for faithful characters if applicable.
+      perf = (length(filter(i -> degree(tbl[i]) == 1, 1:length(tbl))) == 1)
```
Comment (fingolfin) [inline experimental/OrthogonalDiscriminants/src/data.jl]:
```suggestion
      perf = (count(i -> degree(tbl[i]) == 1, 1:length(tbl)) == 1)
```
or even nicer:
```suggestion
      perf = (count(x -> degree(x) == 1, tbl) == 1)
```

File: experimental/OrthogonalDiscriminants/src/data.jl [hunk 3]
```diff
@@ -112,13 +113,64 @@ function orthogonal_discriminants(tbl::Oscar.GAPGroupCharacterTable)
...
+      if p == 0 && perf && order(tbl) == 2*order(facttbl)
+#TODO: compute the reductions mod p (not dividing the group order)
+        # deal with the faithful characters of a perfect central extension 2.G,
+        # the image of the spinor norm is trivial
+        for i in 1:length(tbl)
+          chi = tbl[i]
```
Comment (fingolfin) [inline experimental/OrthogonalDiscriminants/src/data.jl:155]:
```suggestion
        for chi in tbl
```

File: experimental/OrthogonalDiscriminants/src/data.jl [hunk 4]
```diff
@@ -112,13 +113,64 @@ function orthogonal_discriminants(tbl::Oscar.GAPGroupCharacterTable)
...
+          deg = degree(chi)
+          ind = indicator(chi)
+          if ind == 1 && length(class_positions_of_kernel(chi)) == 1
+            if mod(deg, 4 ) == 0
+              res[i] = "1"
+            elseif mod(deg, 2 ) == 0
```
Comment (fingolfin) [inline experimental/OrthogonalDiscriminants/src/data.jl]:
```suggestion
            if mod(deg, 4) == 0
              res[i] = "1"
            elseif mod(deg, 2) == 0
```
or
```suggestion
            if mod(deg, 4) == 0
              res[i] = "1"
            elseif mod(deg, 2) == 0
```

**Review by fingolfin — APPROVED** [review]

File: experimental/OrthogonalDiscriminants/src/data.jl (hunk 1 above)
Comment (ThomasBreuer) [reply experimental/OrthogonalDiscriminants/src/data.jl:117]: Currently one gets a list with `"?"` in some places, for the applications (in particular displaying) this is better than an error.

Comment (fingolfin) [reply experimental/OrthogonalDiscriminants/src/data.jl:117]: Perfect. Just wanted to make sure :-)

File: experimental/OrthogonalDiscriminants/src/data.jl (hunk 3 above)
Comment (ThomasBreuer) [reply experimental/OrthogonalDiscriminants/src/data.jl:155]: I need `i` in the assignments.

**Pull request merged**